### PR TITLE
EVA-3525 update validation report

### PR DESCRIPTION
**Pull request by nitin-ebi**
1. Warning counts removed from VCF Validtion:
warning removed
2. Sample Name check  -  section heading update
Sample section heading update
3. Project section added with files mapping
project section with files mapping
4. Gaps increased between 2 collapsible sections and also between collapsible section and the next section
    Also added icon for showing list is expandable 
increased gap

### CODE REVIEW

File: eva_sub_cli/jinja_templates/fasta_check.html [hunk 1]
```diff
@@ -35,22 +36,25 @@
...
             {% endif %}
         {% else %}
             <!-- has possible assemblies but no metadata assembly -->
+            {% set expand_icon = "&#9654;" %}
             {% set icon = "&#10060;" %}
             {% set row_class = "report-section fail collapsible" %}
             {% set text = "No assembly accession found in metadata" %}
         {% endif %}
-        <div class='{{ row_class }}'>{{ icon }} {{ text }} </div>
+        <div class='{{ row_class }}'><span class="icon">{{ expand_icon }}</span> {{ icon }} {{ text }} </div>
```
Comment (apriltuesday) [inline eva_sub_cli/jinja_templates/fasta_check.html:57]: Could the class be called `expand-icon` so it's not confused with the other icon?

File: eva_sub_cli/jinja_templates/project_details.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{% macro project_details_report(project_title, validation_date, submission_dir, vcf_fasta_analysis_mapping) -%}
+    <div class="report-section info">
+          {% if project_title %}
+             <p><b>Project Title:</b> {{ project_title }}</p>
+         {% else %}
+             <p><b>Project Title:</b></p>
```
Comment (apriltuesday) [inline eva_sub_cli/jinja_templates/project_details.html:6]: Maybe omit this or put "not found" if the title is missing (though I think it's required in the metadata so should probably be parsed out if possible)

File: eva_sub_cli/validators/validator.py [hunk 3]
```diff
@@ -522,8 +522,40 @@ def _collect_md5sum_to_metadata(self):
...
+                        if vcf_file in vcf_analysis_dict:
+                            vcf_fasta_mapping.update({'analysis': vcf_analysis_dict.get(vcf_file)})
+                        else:
+                            vcf_fasta_mapping.update({'analysis': 'Could not be linked'})
+
+                    return vcf_fasta_analysis_mapping
+                except Exception as e:
+                    self.error('Error building Validation Report : Error getting info from metadata file' + str(e))
+        else:
+            self.error('Error building validation report : Metadata file not present')
```
Comment (apriltuesday) [inline eva_sub_cli/validators/validator.py:553]: I think this does almost exactly the same thing as [this logic](https://github.com/EBIvariation/eva-sub-cli/blob/main/bin/check_fasta_insdc.py#L142) in the fasta check. It should be possible to either extract and re-use that logic, or adjust the results from that check and then use those directly in the report (currently looks like [this](https://github.com/EBIvariation/eva-sub-cli/blob/main/tests/test_report.py#L104) - I think it just needs the vcf file listing).

If we're worried about the fasta check not being run but still wanting to include the summary, I guess code reuse is the best bet. But #38 should help make that one more reliable anyway.

**Review by apriltuesday — COMMENTED** [review]
Some implementation suggestions, the HTML report looks way better though :+1:

File: eva_sub_cli/jinja_templates/sample_name_check.html [hunk 4]
```diff
@@ -3,13 +3,16 @@
...
     {% for analysis, results_for_analysis in results.get('results_per_analysis', {}).items() %}
         {% if results_for_analysis.get('difference') %}
+            {% set expand_icon = "&#9654;" %}
             {% set icon = "&#10060;" %}
             {% set row_class = "report-section fail collapsible" %}
+            {% set text = "Sample names in metadata do not matches with those in VCF files" %}
         {% else %}
             {% set icon = "&#10004;" %}
             {% set row_class = "report-section pass" %}
+            {% set text = "Sample names in metadata matches with those in VCF files" %}
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/jinja_templates/sample_name_check.html]: updated

File: eva_sub_cli/validators/validator.py [hunk 5]
```diff
@@ -39,6 +38,7 @@ def __init__(self, mapping_file, submission_dir, metadata_json=None, metadata_xl
                  submission_config: WritableConfig = None):
         # validator write to the validation output directory
         # If the submission_config is not set it will also be written to the VALIDATION_OUTPUT_DIR
+        self.submission_dir = submission_dir
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/validators/validator.py:41]: updated

File: eva_sub_cli/jinja_templates/fasta_check.html (hunk 1 above)
Comment (nitin-ebi) [reply eva_sub_cli/jinja_templates/fasta_check.html]: done

File: eva_sub_cli/jinja_templates/project_details.html (hunk 2 above)
Comment (nitin-ebi) [reply eva_sub_cli/jinja_templates/project_details.html]: updated

**Review by apriltuesday — APPROVED** [review]
As discussed, the refactoring comment is optional and can be done later if needed :+1:

**Review by tcezard — APPROVED** [review]

**Pull request merged**